packages/hierarchies/src/hierarchies/HierarchyNode.ts

```diff
@@ -340,3 +342,19 @@ export type ParsedHierarchyNode<TBase = ParsedCustomHierarchyNode | ParsedInstan
 export type ParsedCustomHierarchyNode = ParsedHierarchyNode<ProcessedCustomHierarchyNode>;
 /** A kind of `ProcessedInstanceHierarchyNode` that has unformatted label and doesn't know about its ancestors. */
 export type ParsedInstanceHierarchyNode = ParsedHierarchyNode<ProcessedInstanceHierarchyNode>;
+
+/**
+ * Properties that are assigned to a processed node if filtering is enabled.
+ * @internal
+ */
+export interface FilteredNodeProps {
+  isFilterTarget?: boolean;
+  hasFilterTargetAncestor?: boolean;
+  filteredChildrenIdentifierPaths?: HierarchyNodeIdentifiersPath[];
+}
+
+/**
+ * A `HierarchyNode` that has been fully processed and only has information, needed for rendering.
+ * @beta
+ */
+export type FinalizedNode = HierarchyNode & FilteredNodeProps;
```

Comment:
I'm not a big fan of this... `HierarchyNode` is already the type that means a fully finalized node, so not sure introducing another one makes much sense. What do you think about adding those props to `BaseHierarchyNode`?

```ts
interface BaseHierarchyNode {
  // ...

  filtering?: {
    isFilterTarget?: boolean;
    hasFilterTargetAncestor?: boolean;
    filteredChildrenIdentifierPaths?: HierarchyNodeIdentifiersPath[];
  };
}

```

Reply:
Made suggested change